explorer/src/providers/accounts/index.tsx

```diff
@@ -108,7 +109,9 @@ export function AccountsProvider({ children }: AccountsProviderProps) {
         <TokensProvider>
           <HistoryProvider>
             <RewardsProvider>
-              <FlaggedAccountsProvider>{children}</FlaggedAccountsProvider>
+              <MetadataProvider>
```

Comment:
Debating the benefit of this. Might just stick a Metadata property on `TokenProgramData`. That way we only attempt to generate a PDA when we need to and can just pass it down through the components on the Account's page.